torch/nn/modules/distance.py

```diff
@@ -1,6 +1,39 @@
-
+import torch
+from .module import Module
 # TODO: Cosine
-# TODO: CosineDistance - make sure lua's CosineDistance isn't actually cosine similarity
+
+
+class CosineDistance(Module):
+    """
+    Computes cosine similarity between two vectors x1,x2:
+
+          x1*x2
+        ---------
+        |x1|*|x2|
+
+    """
+
+    def __init__(self):
+        super(CosineDistance, self).__init__()
+        self.eps = 1e-12
+
+    def forward(self, x):
+        assert len(x) == 2, "Input needs to be two vectors"
+        x1,x2 = x
+        if x1.dim() == 1 or x2.dim() == 1:
```

Comment:
I'm not sure what's the motivation behind that. There's no need to use ops that care about the dimensions, you can multiply any tensors that have the same number of elements, and `torch.norm` can return you a single scalar too.